NWNXLib/Services/PerObjectStorage/PerObjectStorage.cpp

```diff
@@ -33,28 +33,28 @@ PerObjectStorage::ObjectStorage* PerObjectStorage::GetObjectStorage(API::Types::
 }
 
 
-void PerObjectStorage::Set(API::CGameObject *pGameObject, std::string key, int value)
+void PerObjectStorage::Set(API::CGameObject *pGameObject, const std::string& key, int value)
 {
     if (auto *pOS = GetObjectStorage(pGameObject))
     {
         pOS->GetIntMap()[key] = value;
     }
 }
-void PerObjectStorage::Set(API::CGameObject *pGameObject, std::string key, float value)
+void PerObjectStorage::Set(API::CGameObject *pGameObject, const std::string& key, float value)
 {
     if (auto *pOS = GetObjectStorage(pGameObject))
     {
         pOS->GetFloatMap()[key] = value;
     }
 }
-void PerObjectStorage::Set(API::CGameObject *pGameObject, std::string key, std::string value)
+void PerObjectStorage::Set(API::CGameObject *pGameObject, const std::string& key, std::string value)
 {
     if (auto *pOS = GetObjectStorage(pGameObject))
     {
         pOS->GetStringMap()[key] = value;
```

Comment:
Since you're already cleaning this up, might as well move the value in (`emplace()` method, IIRC)

Reply:
Will do!